processor/src/main/java/org/mapstruct/ap/internal/model/StreamMappingMethod.java

```diff
@@ -0,0 +1,404 @@
+                formattingParameters,
+                selectionParameters,
+                sourceRHS,
+                false
+            );
+
+            if ( assignment == null ) {
+
+                assignment = forgeMapping( sourceRHS, sourceElementType, targetElementType );
+
+            }
+            else {
+                if ( method instanceof ForgedMethod ) {
+                    ForgedMethod forgedMethod = (ForgedMethod) method;
+                    forgedMethod.addThrownTypes( assignment.getThrownTypes() );
+
+                }
+            }
+
+            assignment = new Java8FunctionWrapper(
+                assignment,
+                new ArrayList<Type>(),
+                ctx.getTypeFactory().getType( JavaStreamConstants.FUNCTION_FQN )
+            );
+
+            // mapNullToDefault
+            boolean mapNullToDefault = false;
+            if ( method.getMapperConfiguration() != null ) {
+                mapNullToDefault = method.getMapperConfiguration().isMapToDefault( nullValueMappingStrategy );
+            }
+
+            MethodReference factoryMethod = null;
+            if ( !method.isUpdateMethod() ) {
+                factoryMethod = ctx.getMappingResolver().getFactoryMethod( method, method.getResultType(), null );
+            }
+
+            Set<String> existingVariables = new HashSet<String>( method.getParameterNames() );
+            existingVariables.add( loopVariableName );
+
+            List<LifecycleCallbackMethodReference> beforeMappingMethods = LifecycleCallbackFactory.beforeMappingMethods(
+                method,
+                selectionParameters,
+                ctx,
+                existingVariables
+            );
+            List<LifecycleCallbackMethodReference> afterMappingMethods = LifecycleCallbackFactory.afterMappingMethods(
+                method,
+                selectionParameters,
+                ctx,
+                existingVariables
+            );
+
+            Set<Type> helperImports = new HashSet<Type>();
+            if ( resultType.isIterableType() ) {
+                helperImports.add( ctx.getTypeFactory().getType( JavaStreamConstants.COLLECTORS_FQN ) );
+            }
+
+            if ( !sourceParameterType.isCollectionType() && !sourceParameterType.isArrayType() &&
+                sourceParameterType.isIterableType() ) {
+                helperImports.add( ctx.getTypeFactory().getType( JavaStreamConstants.STREAM_SUPPORT_FQN ) );
+            }
+
+            return new StreamMappingMethod(
+                method,
+                assignment,
+                factoryMethod,
+                mapNullToDefault,
+                loopVariableName,
+                beforeMappingMethods,
+                afterMappingMethods,
+                selectionParameters,
+                helperImports,
+                forgedMethod
+            );
+        }
+
+        private Assignment forgeMapping(SourceRHS sourceRHS, Type sourceType, Type targetType) {
+
```

Comment:
Just a small thing: there are some superfluous empty lines like this one here and there.